File: services/nuldoc/nuldoc-src/generators/not_found.ts
import NotFoundPage from "../pages/NotFoundPage.tsx";
import { renderToDOM } from "../jsx/render.ts";
import { Config } from "../config.ts";
import { Page } from "../page.ts";

export type NotFoundPage = Page;

export async function generateNotFoundPage(
  config: Config,
): Promise<NotFoundPage> {
  const html = await renderToDOM(
    NotFoundPage(config),
  );

  return {
    root: html,
    renderer: "html",
    destFilePath: "/404.html",
    href: "/404.html",
  };
}

File: services/nuldoc/lib/nuldoc/generators/not_found.rb
module Nuldoc
  module Generators
    class NotFound
      def initialize(site, config)
        @site = site
        @config = config
      end

      def generate
        html = Pages::NotFoundPage.render(site: @site, config: @config)

        Page.new(
          root: html,
          renderer: :html,
          site: @site,
          dest_file_path: '/404.html',
          href: '/404.html'
        )
      end
    end
  end
end
